src/logfile_values_experimental.py

```diff
@@ -0,0 +1,244 @@
+    parser.add_argument('--file', default='/var/log/messages')
+    parser.add_argument('--metric', type=Metric, nargs='+')
+    parser.add_argument('--time-format', default='%Y-%m-%dT%H:%M:%S')
+    parser.add_argument('--arch', action='store_true')
+    parser.add_argument('--debug', '-d', action='store_true')
+    parser.add_argument('--time-zone', default='UTC')
+    return parser.parse_args()
+
+
+def yield_metrics_values(line, metrics, time_format):
+    values = []
+    fields = line.split()
+    fields[0] = convert_to_timestamp(fields[0], time_format)
+    for metric in metrics:
+        if fields[0] > metric.now - metric.get_timeshift():
+            values.append(int(fields[metric.column]))
+        else:
+            return None
+    return values
+
+
+def get_metrics_last_value(line, metrics):
+    fields = line.split()
+    for metric in metrics:
+        metric.last_value = int(fields[metric.column])
+
+
+def read_logfile_reverse(filename, time_format, metrics, buf_size=8192):
+    """
+    A generator that returns the lines of a file in reverse order.
+    Stops to read when has met timestamp bigger then desired period 
+    
+    returns False if file was not read completely
+    """
+    with open(filename) as fh:
+        global_index = 0
+        segment = None
+        offset = 0
+        fh.seek(0, os.SEEK_END)
+        file_size = remaining_size = fh.tell()
+        while remaining_size > 0:
+            offset = min(file_size, offset + buf_size)
+            fh.seek(file_size - offset)
+            buffer = fh.read(min(remaining_size, buf_size))
+            remaining_size -= buf_size
+            lines = buffer.splitlines()
+            # the first line of the buffer is probably not a complete line so
+            # we'll save it and append it to the last line of the next buffer
+            # we read
+            if segment is not None:
+                # if the previous chunk starts right from the beginning of line
+                # do not concact the segment to the last line of new chunk
+                # instead, yield the segment first
+                if buffer[-1] is not '\n':
+                    lines[-1] += segment
+                else:
+                    yield yield_metrics_values(segment, metrics, time_format)
+            segment = lines[0]
+            for index in range(len(lines) - 1, 0, -1):
+                global_index += 1
+                if lines[index]:
+                    if global_index == 1:
+                        get_metrics_last_value(lines[index], metrics)
+                    yield yield_metrics_values(lines[index], metrics,
+                                               time_format)
+
+    if segment is not None:
+        yield yield_metrics_values(segment, metrics, time_format)
+
+
+def convert_to_timestamp(time_str, time_format):
+    try:
+        timestamp = int(
+            time.mktime(time.strptime(time_str.split("+")[0], time_format)))
```

Comment:
You used single quotes on other places.  Let's be consistent.

Reply:
`split('+', 1)` if you don't need the elements after the first one.

Reply:
You sure it`s anyway returning list :-/

Reply:
Sure.  The second argument of `split()` is to tell Python to stop after some occurrences.  It is good practice to specify it when you are interested in only some elements in the beginning.